### PR TITLE
Update to rust-2015-04-02

### PR DESCRIPTION
**Pull request by alexcrichton**
Along the way, this also removes the ability to specify `[[lib]]` targets while
getting warnings. While we're at it remove nearly all unstable features that
Cargo is depending on as well!

r? @huonw

### CODE REVIEW

File: src/bin/cargo.rs [hunk 1]
```diff
@@ -153,18 +153,16 @@ fn execute(flags: Flags, config: &Config) -> CliResult<Option<()>> {
 }
 
 fn find_closest(cmd: &str) -> Option<String> {
-    match list_commands().iter()
-                            // doing it this way (instead of just .min_by(|c|
```
Comment (huonw) [inline src/bin/cargo.rs]: Does this comment still apply?

File: src/cargo/ops/cargo_rustc/custom_build.rs [hunk 2]
```diff
@@ -256,7 +256,7 @@ impl BuildOutput {
         let whence = format!("build script of `{}`", pkg_name);
 
         for line in input.lines() {
-            let mut iter = line.splitn(1, |c| c == ':');
+            let mut iter = line.splitn(2, |c| c == ':');
```
Comment (huonw) [inline src/cargo/ops/cargo_rustc/custom_build.rs]: s/`|c| c == ':'`/`':'`/?

File: src/cargo/ops/cargo_rustc/custom_build.rs [hunk 3]
```diff
@@ -267,7 +267,7 @@ impl BuildOutput {
             };
 
             // getting the `key=value` part of the line
-            let mut iter = data.splitn(1, |c| c == '=');
+            let mut iter = data.splitn(2, |c| c == '=');
```
Comment (huonw) [inline src/cargo/ops/cargo_rustc/custom_build.rs]: Similarly.

**Comment by alexcrichton** [comment]
Pushed a few updates, thanks for taking a look @huonw!

**Pull request merged**